psi4/src/psi4/libmints/twobody.cc

```diff
@@ -112,6 +116,61 @@ TwoBodyAOInt::TwoBodyAOInt(const TwoBodyAOInt &rhs) : TwoBodyAOInt(rhs.integral_
 
 TwoBodyAOInt::~TwoBodyAOInt() {}
 
+// Haser 1989, Equation 7 
+void TwoBodyAOInt::update_density(const std::vector<SharedMatrix>& D) {
+    
+    timer_on("Update Density");
+#pragma omp parallel for
+    for (int M = 0; M < nshell_; M++) {
```

Comment:
Why are these loops independent? This code counts (M, N) and (N, M). I believe it would be more efficient to only count one.